draft-barnes-acme.md

```diff
@@ -536,121 +508,101 @@ Recovery tokens are employed in response to Recovery Challenges.  Such challenge
 
 ~~~~~~~~~~
 
-{
-  "type": "certificate",
-  "certificate": "Zmzdx7UKvwDJ6bk...YBX22NPGQZyYcg",
-  "chain": [
-    "WUn8L2vLT553pIWJ2...gJ574o2anls1k2p",
-    "y3O4puZa9r5KBk1LX...Ya7jlaAZUfuYZGZ"
-  ],
-  "refresh": "https://example.com/refresh/Dr8eAwTVQfSS/"
-}
+HTTP/1.1 201 Created
+Content-Type: application/pkix-cert
+Link: <https://example.com/acme/ca-cert>;rel="up"
+Location: https://example.com/acme/cert/asdf
+
+[DER-encoded certificate]
 
 ~~~~~~~~~~
 
-The certificate message allows the server to provide the certificate itself, as well as some associated management information.  The chain of CA certificates can simplify TLS server configuration, by allowing the server to suggest the certificate chain that a TLS server using the issued certificate should present.
+## Certificate Refresh
 
-The refresh URI allows the client to download updated versions of the issued certificate, in the sense of certificates with different validity intervals, but otherwise the same contents (in particular, the same names and public key).  This can be useful in cases where a CA wishes to issue short-lived certificates, but is still willing to vouch for an identifier-key binding over a longer period of time.  To download an updated certificate, the client simply sends a GET request to the refresh URI.
+The certificate URL (provided in the Location header of the server's response) is used to refresh or revoke the certificate.  To refresh the certificate, the client simply sends a GET request to the certificate URL.  This allows the server to provide the client with updated certificates with the same content and different validity intervals, for as long as the server considers the client authorized.
 
+If a client sends a refresh request and the server is not willing to refresh the certificate, the server MUST respond with status code 403 (Forbidden).  If the client still wishes to obtain a certificate, it can re-initiate the authorization process for any expired authorizations related to the certificate.
 
 ## Certificate Revocation
 
-To request that a certificate be revoked, the client sends a revocationRequest message that indicates the certificate to be revoked, with a signature by an authorized key:
-
-type (required, string):
-: "revocationRequest"
-
-certificate (required, string):
-: The certificate to be revoked.
+To request that a certificate be revoked, the client sends a POST request to the certificate URL.  The body of the POST is a JWS object whose JSON payload contains an indication when the client would like the certificate to be revoked:
 
-signature (required, object):
-: A signature object reflecting a signature by an authorized key pair over the certificate.
+revoke (required, string):
+: The time at which the certificate should be revoked.  The value of this field MUST be either the literal string "now", or a date in RFC 3339 format {{RFC3339}}.
```

Comment:
If there is a use case for anything other than "now", then allow just the date.  Otherwise, "none" is implied, and there is no data to be conveyed here (so `null`, I guess).